### PR TITLE
UPI refinements

### PR DESCRIPTION
**Pull request by goergisn**
### `FormTextInputItemView`
- Report `false` of `canBecomeFirstResponder` when item is hidden
- Only `becomeFirstResponder` when the item is not hidden
### `FormViewController`
- Only assign first responder if the item is not hidden (So all items adopt the behavior automatically)
- Exposed `assignInitialFirstResponder` as `@_spi(AdyenInternal)` to allow the `UPIComponent` to re-assign focus to the VPA input field when switching flows 
  - ⚠️ Not sure about this one, maybe @erenbesel you have a better idea?)
### `FormSegmentedControlItem`
- Actually applying the provided text style
### `UPIComponent`
- Simplifying code

### CODE REVIEW

File: Adyen/Core/Payment Methods/UPIPaymentMethod.swift [hunk 1]
```diff
@@ -14,7 +14,7 @@ public struct UPIPaymentMethod: PaymentMethod {
     public let name: String
 
     /// The available UPI apps.
-    public let apps: [Issuer]
+    public let apps: [Issuer]?
```
Comment (nauaros) [inline Adyen/Core/Payment Methods/UPIPaymentMethod.swift:17]: Why is this optional now? Shouldn't there be at least one issuer in order to initialize the payment method?

File: Adyen/UI/Form/FormViewController.swift [hunk 2]
```diff
@@ -228,9 +228,10 @@ open class FormViewController: UIViewController, AdyenObserver, PreferredContent
 
     // MARK: - Other
 
-    private func assignInitialFirstResponder() {
+    @_spi(AdyenInternal)
+    public func assignInitialFirstResponder() {
         guard view.isUserInteractionEnabled else { return }
-        let textItemView = itemManager.topLevelItemViews.first(where: { $0.canBecomeFirstResponder })
+        let textItemView = itemManager.topLevelItemViews.first(where: { $0.canBecomeFirstResponder && !$0.isHidden })
```
Comment (nauaros) [inline Adyen/UI/Form/FormViewController.swift:233]: nice. good catch

File: Adyen/Core/Payment Methods/UPIPaymentMethod.swift (hunk 1 above)
Comment (goergisn) [reply Adyen/Core/Payment Methods/UPIPaymentMethod.swift:17]: apps are optional - if the list is empty we show the VPA input field - so there is always one item in the list but the last item is always the VPA input field (not an app)

File: Tests/SnapshotTests/Components/UPIComponentUITests.swift [hunk 3]
```diff
@@ -145,29 +156,30 @@ class UPIComponentUITests: XCTestCase {
...
-
-        wait(for: .milliseconds(300))
         
-        sut.currentSelectedIndex = 0
-        sut.currentSelectedItem = sut.upiAppsList.first
-        sut.currentSelectedItem?.isSelected = true
-        sut.virtualPaymentAddressItem.isHidden.wrappedValue = true
+        sut.upiAppsList.first?.selectionHandler?()
 
+        // TODO: Wait until the first item is selected
```
Comment (nauaros) [inline Tests/SnapshotTests/Components/UPIComponentUITests.swift]: Forgotten or gonna be followed up in a second PR?

**Review by nauaros — APPROVED** [review]

File: Tests/SnapshotTests/Components/UPIComponentUITests.swift (hunk 3 above)
Comment (goergisn) [reply Tests/SnapshotTests/Components/UPIComponentUITests.swift]: Yesss forgotten - removed it 👌

File: Adyen/UI/Form/FormViewController.swift [hunk 4]
```diff
@@ -228,9 +228,10 @@ open class FormViewController: UIViewController, AdyenObserver, PreferredContent
 
     // MARK: - Other
 
-    private func assignInitialFirstResponder() {
+    @_spi(AdyenInternal)
+    public func assignInitialFirstResponder() {
```
Comment (atmamont) [inline Adyen/UI/Form/FormViewController.swift]: why do we need to make it public, is this for tests?

**Review by atmamont — APPROVED** [review]

File: Adyen/UI/Form/FormViewController.swift (hunk 4 above)
Comment (goergisn) [reply Adyen/UI/Form/FormViewController.swift]: Check out https://github.com/Adyen/adyen-ios/pull/1711#issue-2349160205

It's exposed to re-assign focus to the text field when switching back from the QR code tab

**Pull request merged**